src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -153,15 +170,9 @@ public static bool IsRetryableWriteException(Exception exception)
                         case ServerErrorCode.HostUnreachable:
                         case ServerErrorCode.NetworkTimeout:
                         case ServerErrorCode.SocketException:
+                        case ServerErrorCode.ExceededTimeLimit:
```

Comment:
This change is taken from https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/106